app/src/org/commcare/android/fragments/SetupEnterURLFragment.java

```diff
@@ -0,0 +1,97 @@
+package org.commcare.android.fragments;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.support.v4.app.Fragment;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import org.commcare.android.framework.ManagedUi;
+import org.commcare.android.framework.UiElement;
+import org.commcare.dalvik.R;
+import org.javarosa.core.services.locale.Localization;
+
+/**
+ * Fragment for inputting app installation URL, "returned" through the URLInstaller interface.
+ * Created by dancluna on 3/17/15.
+ */
+@ManagedUi(R.layout.setup_enter_url)
+public class SetupEnterURLFragment extends Fragment {
+
+    public interface URLInstaller {
+        /**
+         * Called when user fills in an URL and presses 'Start Install'.
+         * The parent activity is responsible for implementing this interface and doing something with the URL.
+         * @param url URL typed by the user
+         */
+        public void OnURLChosen(String url);
+    }
+
+    public static final String interfaceName = URLInstaller.class.getName();
+
+    private URLInstaller listener;
+
+    @UiElement(R.id.start_install)
+    Button installButton;
+
+    @UiElement(R.id.url_spinner)
+    Spinner prefixURLSpinner;
+
+    @UiElement(R.id.edit_profile_location)
+    EditText profileLocation;
+
+    @Override
+    public View onCreateView(final LayoutInflater inflater, final ViewGroup container, final Bundle savedInstanceState) {
+        final View view = inflater.inflate(R.layout.setup_enter_url, container, false);
+        installButton = (Button) view.findViewById(R.id.start_install);
+        installButton.setText(Localization.get("install.button.start"));
+        prefixURLSpinner = (Spinner) view.findViewById(R.id.url_spinner);
+        profileLocation = (EditText) view.findViewById(R.id.edit_profile_location);
+        final TextView appProfile = (TextView) view.findViewById(R.id.TextView02);
+        appProfile.setText(Localization.get("install.appprofile"));
+
+        installButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(final View v) {
+                getFragmentManager().popBackStack();
+                if(listener != null) listener.OnURLChosen(getURL());
```

Comment:
onAttach is always called before onCreateView, so you don't need to provide the listener null check here as long as the fragment will fail to attach unless its parent is of the right type.